java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/aggregation/FirstLastValueAggFuncWithoutRetract.java

```diff
@@ -69,13 +78,29 @@ public void merge(FirstLastValueAccumulator<T> target, FirstLastValueAccumulator
     }
 
     @Override
-    public void retract(FirstLastValueAccumulator<T> accumulator, T value) {
+    public void retract(AbstractRawDataAggFunc.RawDataAccumulator<T> accumulator, T value) {
         throw new UnsupportedOperationException();
     }
 
     @Override
     public void retractAccumulator(
-            FirstLastValueAccumulator<T> target, FirstLastValueAccumulator<T> source) {
+            AbstractRawDataAggFunc.RawDataAccumulator<T> target,
+            AbstractRawDataAggFunc.RawDataAccumulator<T> source) {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public DataType getResultDatatype() {
+        return inDataType;
+    }
+
+    @Override
+    public AbstractRawDataAggFunc.RawDataAccumulator<T> createAccumulator() {
```

Comment:
We need to make sure that `FirstLastValueAggFuncWithoutRetract` and `FirstLastValueAggFunc` use the same accumulator type, so that SlidingWindowKeyedProcessFunction can handle the accumulators passed from the pre-agg process.